silkworm/db/snapshots/seg/compressor/pattern_aggregator.cpp

```diff
@@ -57,6 +57,15 @@ class PatternAggregatorImpl {
     vector<Pattern> aggregate();
 
   private:
+    /**
+     * The max number of patterns to keep in reserve
+     * while building the sorted list incrementally in order_by_score_and_limit.
+     * Having more than kMaxPatterns patterns in consideration
+     * produces 10% better compression ratio while still keeping RAM usage under control.
+     * see DictReducerSoftLimit in erigon
+     */
+    static constexpr size_t kMaxPatternsSoftLimit = 1'000'000;
```

Comment:
Can we have kMaxPatternsSoftLimit value referring to kMaxPatterns? E.g:
```
static constexpr size_t kMaxPatternsSoftLimit = kMaxPatterns * 16;
```

Reply:
this constant has to be identical to erigon's in order to produce bit-identical .seg files (having the same torrent hash to be seeded together with erigon 3 instances)